### PR TITLE
Add .odoignore documentation

### PR DESCRIPTION
**Pull request by cdrage**
Adds .gitignore documentation.

Closes https://github.com/redhat-developer/odo/issues/1427

### CODE REVIEW

**Comment by cdrage** [comment]
Not too sure if this is enough @kadel

Or do you suggest separate documentation?

**Comment by cdrage** [comment]
Yeah you're right, one line of code depicts that: https://github.com/redhat-developer/odo/blob/f816d9f41ae2d3be64bd5b28e7fa841bc48f3c77/pkg/util/util.go#L403

I'll update the documentation.

File: docs/getting-started.md [hunk 1]
```diff
@@ -237,6 +237,28 @@ After any modification to your shell configuration file, you will need to `sourc
...
+
+```ini
+.git
+*.js
+/tests
+```
+
+Will ignore `.git` files, any files with the `.js` extension and the folder `tests`.
+
+`.odoignore` allows any [regular expression / regex](https://regexr.com/), for example:
```
Comment (mik-dass) [inline docs/getting-started.md]: No not regular/ regrex expressions, glob expressions.

File: docs/getting-started.md [hunk 2]
```diff
@@ -237,6 +237,28 @@ After any modification to your shell configuration file, you will need to `sourc
 **NOTE**: The completion system will stop working if you either rename the `odo` executable or move it. You will therefore need
 to re-enable it accordingly.
 
+## Using the `.odoignore` and `.gitignore` files
+
+The `.odoignore` file in the root directory of your application will ignore a list of files/patterns. This will prevent `odo watch` from constantly reloading / redeploying certain files.
```
Comment (mik-dass) [inline docs/getting-started.md]: `odo push` uses .odoignore too, so I guess that should be mentioned too

**Comment by cdrage** [comment]
Thanks @mik-dass I've updated this based upon your suggestions.

**Comment by cdrage** [comment]
Can anyone else please LGTM it? :+1: @anmolbabu @surajnarwade @girishramnani @amitkrout @kadel

**Pull request merged**